src/UserSpaceInstrumentation/AllocateInTracee.cpp

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "AllocateInTracee.h"
+
+#include <absl/base/casts.h>
+#include <absl/strings/numbers.h>
+#include <absl/strings/str_format.h>
+#include <absl/strings/str_split.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <sys/ptrace.h>
+#include <sys/wait.h>
+
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "OrbitBase/SafeStrerror.h"
+#include "UserSpaceInstrumentation/RegisterState.h"
+
+namespace orbit_user_space_instrumentation {
+
+namespace {
+
+using orbit_base::ReadFileToString;
+
+// Returns the address range of the first executable memory region. In every case I encountered this
+// was the second line in the `maps` file corresponding to the code of the process we look at.
```

Comment:
this is only true for static executables for dynamic executables it is entirely possible that the region belong to linker.

I would check for executable name here if you are looking for the main executable. (resolve /proc/pid/exe and compare file names)